test/ETO/Extras/ExtraTokenControllers.js

```diff
@@ -0,0 +1,352 @@
+          const tokenAllowance = await equityToken.allowance(
+            investor1,
+            equityTokenController.address,
+          );
+          expect(tokenAllowance).to.be.bignumber.eq(inv1DistAmount);
+          // check if forced transfer is allowed
+          let isAllowed = await equityTokenController.onTransfer(
+            equityTokenController.address,
+            investor1,
+            investor2,
+            inv1DistAmount,
+          );
+          expect(isAllowed).to.be.true;
+          // different amount is not allowed
+          isAllowed = await equityTokenController.onTransfer(
+            equityTokenController.address,
+            investor1,
+            investor2,
+            inv1DistAmount.sub(1),
+          );
+          expect(isAllowed).to.be.false;
+          await equityTokenController.executeForcedTransfer(investor1);
+          expect(await equityToken.balanceOf(investor1)).to.be.bignumber.eq(0);
+          expect(await equityToken.balanceOf(investor2)).to.be.bignumber.eq(inv1DistAmount);
+          // forced transfer cannot be executed again
+          await expect(equityTokenController.executeForcedTransfer(investor1)).to.be.rejectedWith(
+            "NF_FORCED_T_NOT_EXISTS",
+          );
+        });
+
+        it("should freeze and unfreeze account", async () => {
+          // make successful transfer
+          await equityToken.transfer(investor2, 1, { from: investor1 });
+          // freeze investor1
+          await equityTokenController.freezeHolder(investor1, { from: company });
+          // no permission to send
+          let canTransfer = await equityTokenController.onTransfer(
+            investor1,
+            investor1,
+            investor2,
+            "1",
+          );
+          expect(canTransfer).to.be.false;
+          await expect(equityToken.transfer(investor2, 1, { from: investor1 })).to.be.rejectedWith(
+            EvmError,
+          );
+          // no permission to receive
+          canTransfer = await equityTokenController.onTransfer(
+            investor2,
+            investor2,
+            investor1,
+            "1",
+          );
+          expect(canTransfer).to.be.false;
+          await expect(equityToken.transfer(investor1, 1, { from: investor2 })).to.be.rejectedWith(
+            EvmError,
+          );
+          // no permission to send via broker
+          canTransfer = await equityTokenController.onTransfer(
+            investor1,
+            investor1,
+            investor2,
+            "1",
+          );
+          expect(canTransfer).to.be.false;
+          // todo: test forced transfer interaction, but that's for the real code
+          // now unfreeze
+          await equityTokenController.unfreezeHolder(investor1, { from: company });
+          // receive
+          await equityToken.transfer(investor2, 1, { from: investor1 });
+          // send
+          await equityToken.transfer(investor1, 1, { from: investor2 });
```

Comment:
I would prefer to have actual assert-statements here, not just executed transfers: e.g.
```
          await equityToken.transfer(investor2, 1, { from: investor1 });
          expect(await equityToken.balanceOf(investor2)).to.be.bignumber.eq(2);
          // send
          await equityToken.transfer(investor1, 1, { from: investor2 });
          expect(await equityToken.balanceOf(investor2)).to.be.bignumber.eq(1);
```